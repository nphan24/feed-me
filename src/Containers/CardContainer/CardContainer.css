.card-container {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
}

.card-container-section {
  background-color: #abd8ea;
  height: 100vh;
}

.loading-comment {
  margin-top: 100px;
  font-size: 3em;
  text-align: center;
}

.loading-gif {
  width: 200px;
  height: 180px;
}