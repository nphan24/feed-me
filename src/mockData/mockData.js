/* eslint-disable */

export const mockRecipe = {
  name: 'pasta',
  image: 'url',
  totalTime: '30',
  id: 2
};

export const uncleanedMockRecipeData = { matches: [
  { recipeName: 'tofu', 
    imageUrlsBySize: 'url', 
    ingredients: ['',''], 
    totalTimeInSeconds: 3600, 
    id: 2,
    pasta: 'speghetti',
    fruit: 'mango'
  }
]
 };

export const cleanedMockRecipeData = [{ 
  Ingredients: ['', ''], 
  id: 2, 
  image: undefined, 
  name: 'tofu', 
  totaltime: 60 
}];

export const mockRecipesArray = [
  {
    name: 'pasta',
    image: 'url',
    totalTime: '30',
    id: 2
  },
  {
    name: 'rice',
    image: 'hmm',
    totalTime: '90',
    id: 3
  }
];

export const mockRemainingArray = [
  {
    name: 'rice',
    image: 'hmm',
    totalTime: '90',
    id: 3
  }
];
