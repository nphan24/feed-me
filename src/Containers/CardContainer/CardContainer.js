import React from 'react';
import { connect } from 'react-redux';
import { Route, withRouter, Switch } from 'react-router-dom';
import * as routes from '../../constants/routes';
import Card from '../Card/Card';
import './CardContainer.css';
import Login from '../Login/Login';
import SignUp from '../SignUp/SignUp';
import PropTypes from 'prop-types';
import PageChange from '../PageChange/PageChange';

const loading = require('../../assets/loading.gif');
const noFavorites = require('../../assets/favorites.gif');

export const CardContainer = props => {

  const cardType = 
    props.location.pathname === '/favorites' ? 'favorites': 'recipes';

  let renderCards;
  
  if (props.recipes.length === 0) {
    renderCards = (
      <div>
        <p className='loading-comment'>
          Loading...
        </p>
        <img className='loading-gif'src={loading} alt='loading'/>
      </div>
    );
  } else if (props.location.pathname === '/favorites' && 
    props.favorites.length === 0) {
    renderCards = <div>
      <p className="loading-comment">There are no favorites!</p>
      <img className="nofavorites-gif" src={noFavorites} alt="no favorites" />
    </div>;
  } else { 
    renderCards = props[cardType].map((recipe, index) => {

      return (
        <Card 
          key={recipe.name + index}
          recipe={recipe}
        />
      );
    });
  }

  return <section className="card-container-section">
    <Switch>
      <Route 
        exact path="/" 
        render={() => {
          return (
            <div className="card-container">
              {renderCards}
            </div>);
        }}/>
      <Route 
        exact path="/favorites" 
        render={() => {
          return (
            <div className="card-container">
              {renderCards}
            </div>);
        }} />
      <Route exact path={routes.LOGIN} component={Login} />
      <Route exact path={routes.SIGN_UP} component={SignUp} />
    </Switch>
    {props.location.pathname === '/' &&
    <PageChange />
    }
  </section>;
};

export const mapStateToProps = state => ({
  recipes: state.recipes,
  favorites: state.favorites
});

CardContainer.propTypes = {
  location: PropTypes.object,
  recipes: PropTypes.array,
  favorites: PropTypes.array
};

export default withRouter(connect(mapStateToProps)(CardContainer));