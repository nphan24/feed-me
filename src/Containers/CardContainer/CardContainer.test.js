import CardContainer from '../CardContainer/CardContainer';
import * as mock from '../../mockData/mockData';

describe('CardDontainer', () => {
  it('should match the snapshot', () => {

  });

  
});

describe('mapStateToProps', () => {

});