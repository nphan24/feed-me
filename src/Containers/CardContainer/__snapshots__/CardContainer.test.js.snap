// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDontainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    favorites={
        Array [
            Object {
              "id": 2,
              "image": "url",
              "name": "pasta",
              "totalTime": "30",
            },
          ]
    }
    location={
        Object {
            "pathname": "favorites",
          }
    }
    recipes={
        Array [
            Object {
              "id": 2,
              "image": "url",
              "name": "pasta",
              "totalTime": "30",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Switch>
          <Route
                    exact={true}
                    path="/"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/favorites"
                    render={[Function]}
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/login"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/signup"
          />
</Switch>,
        false,
      ],
      "className": "card-container-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/favorites"
              render={[Function]}
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/login"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/signup"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/favorites",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/login",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/signup",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      false,
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Switch>
            <Route
                        exact={true}
                        path="/"
                        render={[Function]}
            />
            <Route
                        exact={true}
                        path="/favorites"
                        render={[Function]}
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/login"
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/signup"
            />
</Switch>,
          false,
        ],
        "className": "card-container-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/"
                render={[Function]}
/>,
              <Route
                exact={true}
                path="/favorites"
                render={[Function]}
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/login"
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/signup"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/favorites",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/login",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/signup",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        false,
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
